### PR TITLE
Soen114 charts date picker

### PR DESCRIPTION
**Pull request by amqkv**
# Description

Implements/fixes [SOEN-114](https://soen390team8.atlassian.net/jira/software/projects/SOEN/boards/1?selectedIssue=SOEN-114)

- Added date picker for charts on doctor's dashboard

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit test

# Screenshots

|         Original          |         Updated          |
| :-----------------------: | :----------------------: |
|image | image |

### CODE REVIEW

**Review by stevenvuong00 — APPROVED** [review]
great work! awseome feature

File: frontend/components/doctor/patient-charts-overview.tsx [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
 import { extractStatuses, transformWeightTempData } from "@frontend/functions/data-transform-chart";
 import { Patient } from "@frontend/models/patient";
 import { DAY } from "@frontend/utils/constants";
 import PieChart from "@frontend/components/pie-chart";
 import Table from "@frontend/components/doctor/patient-overview-table";
 import ScatterChart from "@frontend/components/scatter-chart";
+import { CSSProperties, useState } from "react";
 
 export default function PatientChartsOverview({ patientList }: { patientList: Patient[] }) {
+    const today = new Date().toISOString().slice(0, 10);
```
Comment (NicholasYip) [inline frontend/components/doctor/patient-charts-overview.tsx:11]: I see we are importing/using the DAY constant, so is this still needed?

File: frontend/components/doctor/patient-charts-overview.tsx [hunk 2]
```diff
@@ -66,29 +83,31 @@ export default function PatientChartsOverview({ patientList }: { patientList: Pa
...
                         </TabPanel>
                         <TabPanel>
                             <ScatterChart
                                 statuses={extractStatuses(patientList)}
                                 transformDataFn={transformWeightTempData}
                                 day={DAY.ALL}
                                 title={"Patients' Temperature and Weight of All Time"}
-                                h={260}
+                                h={275}
                                 w={400}
```
Comment (NicholasYip) [inline frontend/components/doctor/patient-charts-overview.tsx:111]: do these set widths and heights look good on mobile?

**Review by NicholasYip — APPROVED** [review]
left some small comments, rest is some high quality work

File: frontend/components/doctor/patient-charts-overview.tsx (hunk 1 above)
Comment (amqkv) [reply frontend/components/doctor/patient-charts-overview.tsx:11]: Yep `today` is in the date form (e.g., `2022-03-03`) while DAY constant contains strings like `"today"`, `"yesterday"`, etc

File: frontend/components/doctor/patient-charts-overview.tsx (hunk 2 above)
Comment (amqkv) [reply frontend/components/doctor/patient-charts-overview.tsx:111]: Pretty bueno if I do say so myself 🥸

**Review by jfv-2000 — APPROVED** [review]
B R A V O C A D O 
giphy

**Comment by amqkv** [comment]
> 



> B R A V O C A D O giphy

**Pull request merged**